### ISSUE TITLE
为什么mmdetecion的FCOS训练出来map不能达到论文中的精度

### ISSUE THREAD

**Issue by zhoushuang66** [open-mmlab/mmdetection#8082]
fcos的训练出来map是0.33，论文中的精度是0.371
base_ = [
    '../_base_/datasets/coco_detection.py',
    '../_base_/schedules/schedule_2x.py', '../_base_/default_runtime.py'
]
# model settings
model = dict(
    type='FCOS',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=False),
        norm_eval=True,
        style='caffe',
        init_cfg=dict(
            type='Pretrained',
            checkpoint='open-mmlab://detectron/resnet50_caffe')),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_output',  # use P5
        num_outs=5,
        relu_before_extra_convs=True),
    bbox_head=dict(
        type='FCOSHead',
        num_classes=80,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        strides=[8, 16, 32, 64, 128],
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='IoULoss', loss_weight=1.0),
        loss_centerness=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)),
    # training and testing settings
    train_cfg=dict(assigner=dict(
 47             type='MaxIoUAssigner',
 48             pos_iou_thr=0.5,
 49             neg_iou_thr=0.4,
 50             min_pos_iou=0,
 51             ignore_iof_thr=-1),
 52         allowed_border=-1,
 53         pos_weight=-1,
 54         debug=False),
 55     test_cfg=dict(
 56         nms_pre=1000,
 57         min_bbox_size=0,
 58         score_thr=0.05,
 59         nms=dict(type='nms', iou_threshold=0.5),
 60         max_per_img=100))
 61 img_norm_cfg = dict(
 62     mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False)
 63 train_pipeline = [
 64     dict(type='LoadImageFromFile'),
 65     dict(type='LoadAnnotations', with_bbox=True),
 66     dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
 67     dict(type='RandomFlip', flip_ratio=0.5),
 68     dict(type='Normalize', **img_norm_cfg),
 69     dict(type='Pad', size_divisor=32),
 70     dict(type='DefaultFormatBundle'),
 71     dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
 72 ]
 73 test_pipeline = [
 74     dict(type='LoadImageFromFile'),
 75     dict(
 76         type='MultiScaleFlipAug',
 77         img_scale=(1333, 800),
 78         flip=False,
 79         transforms=[
 80             dict(type='Resize', keep_ratio=True),
 81             dict(type='RandomFlip'),
 82             dict(type='Normalize', **img_norm_cfg),
 83             dict(type='Pad', size_divisor=32),
 84             dict(type='ImageToTensor', keys=['img']),
 85             dict(type='Collect', keys=['img']),
 86         ])
 87 ]
 88    data = dict(
 89     samples_per_gpu=20,
 90     workers_per_gpu=2,
 91     train=dict(pipeline=train_pipeline),
 92     val=dict(pipeline=test_pipeline),
 93     test=dict(pipeline=test_pipeline))
 94 # optimizer
 95 optimizer = dict(
 96     lr=0.01, paramwise_cfg=dict(bias_lr_mult=2., bias_decay_mult=0.))
 97 optimizer_config = dict(
 98     _delete_=True, grad_clip=dict(max_norm=35, norm_type=2))
 99 # learning policy
100 #lr_config = dict(
101 #    policy='step',
102 #    warmup='constant',
103 #    warmup_iters=500,
104 #    warmup_ratio=1.0 / 3,
105 #    step=[8, 11])
106 #runner = dict(type='EpochBasedRunner', max_epochs=12)
107 lr_config = dict(
108     policy='step',
109     warmup='linear',
110     warmup_iters=500,
111     warmup_ratio=0.001,
112     step=[16, 22])
113 runner = dict(type='EpochBasedRunner', max_epochs=24)

**Comment by Czm369** [comment]
Maybe you can try 4 GPUs, and put 4 images on each GPU for training, the accuracy should be improved.

**Comment by zhoushuang66** [comment]
好的，谢谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "open-mmlab/mmdetection"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月30日(星期一) 下午3:45
***@***.***&gt;;
***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [open-mmlab/mmdetection] 为什么mmdetecion的FCOS训练出来map不能达到论文中的精度 (Issue #8082)





 
Maybe you can try 4 GPUs, and put 4 images on each GPU for training, the accuracy should be improved.
